algo/src/main/java/org/neo4j/graphalgo/impl/metaPathComputation/MetaPathPrecomputeHighDegreeNodes.java

```diff
@@ -0,0 +1,318 @@
+        }
+    }
+
+    public void computeMetaPathFromNodeLabel(int nodeID, int metaPathLength) {
+        duplicateFreeMetaPaths.put(nodeID, new HashMap<>());
+        ArrayList<Integer> initialMetaPath = new ArrayList<>();
+        initialMetaPath.add(arrayGraphInterface.getLabel(nodeID));
+        //TODO less hacky
+        HashSet<Integer> instanceHS = new HashSet<>();
+        instanceHS.add(nodeID);
+        computeMetaPathFromNodeLabel(initialMetaPath, instanceHS, metaPathLength - 1);
+    }
+/*
+    private HashSet<Integer> initInstancesRow(int startNodeLabel) {
+        int startNodeLabelId = labelDictionary.get(startNodeLabel);
+        HashSet<Integer> row = initialInstances.get(startNodeLabelId);
+        return row;
+    }*/
+
+    //TODO------------------------------------------------------------------------------------------------------------------
+    public Stream<org.neo4j.graphalgo.impl.metaPathComputation.ComputeAllMetaPaths.Result> resultStream() {
+        return IntStream.range(0, 1).mapToObj(result -> new org.neo4j.graphalgo.impl.metaPathComputation.ComputeAllMetaPaths.Result(new HashSet<>()));
+    }
+
+    @Override
+    public MetaPathPrecomputeHighDegreeNodes me() {
+        return this;
+    }
+
+    @Override
+    public MetaPathPrecomputeHighDegreeNodes release() {
+        return null;
+    }
+
+    /**
+     * Result class used for streaming
+     */
+    public static final class Result {
+
+        HashMap<Integer, HashMap<String, HashSet<Integer>>> finalMetaPaths;
+
+        public Result(HashMap<Integer, HashMap<String, HashSet<Integer>>> finalMetaPaths) {
+            this.finalMetaPaths = finalMetaPaths;
+        }
+
+        @Override
+        public String toString() {
+            return "Result{}";
+        }
+
+        public HashMap<Integer, HashMap<String, HashSet<Integer>>> getFinalMetaPaths() {
+            return finalMetaPaths;
+        }
+    }
+
+    public void weight(int index, int weight) throws Exception {
+        if (weight <= 0 || weight > 10) {
+            throw new Exception("Weight needs to be in range (0;10]");
+        }
+        metaPathsWeights.set(index, weight);
+    }
+
+    private List<Integer> getMaxDegreeNodes() {
+        ArrayList<Integer> list = new ArrayList();
+        List<Integer> maxDegreeNodes;
+        Iterator<Node> nodes;
+        graph.forEachNode(list::add);
+        list.sort(new DegreeComparator(graph)); //TODO Use Array instead of list?
+
+
+        maxDegreeNodes = list.subList((int) (list.size() - Math.ceil((double) list.size() / ratioHighDegreeNodes)), list.size());
+        for (int nodeID : maxDegreeNodes) { //TODO always consecutive? (without gap)
+            debugOut.println("nodeID: " + nodeID + "; degree: " + graph.degree(nodeID, Direction.BOTH) + "; label: " + graph.getLabel(nodeID));
```

Comment:
remove debugOutput if not needed anymore

Reply:
could be useful in the future